Quantum-Control-Applications-QuAM/Superconducting/calibrations/26_two_qubit_xeb.py

```diff
@@ -0,0 +1,62 @@
+import numpy as np
+from quam_libs.components import QuAM, TransmonPair
+from quam_libs.experiments.two_qubit_xeb import (
+    XEBConfig,
+    XEB,
+    backend as fake_backend,
+    QUAGate,
+)
+
+machine = QuAM.load()
+qubits = machine.active_qubits
+# Get the relevant QuAM components
+target_qubit_indices = [3, 4]  # Indices of the target qubits
+target_qubits = [qubits[i] for i in target_qubit_indices]
+target_qubit_pairs = [
+    qubit_pair
+    for qubit_pair in machine.active_qubit_pairs
+    if qubit_pair.qubit_control in target_qubits and qubit_pair.qubit_target in target_qubits
+]
```

Comment:
Hey Arthur, can you help me to understand these lines?

I remember we discussed the need to implement XEB over _distant_ qubit pairs, meaning there might require a sequence of two-qubit gates over qubits that are not "3, 4" in this case.

But on line 18, I believe it will only ever resolve to 3 @ 4 or 4 @ 3. Did you mean:
```python
    if qubit_pair.qubit_control in target_qubits or qubit_pair.qubit_target in target_qubits
                                                 ^^
```
?

Reply:
So here this is a generic way of checking that for an arbitrary number of qubits (no longer restricted to two qubits actually, because the script should now cope with n qubits, I just did not want to advertise it in the README yet because it's not fully tested), all the qubit pairs involve qubits that are already listed in the qubits list. This is necessary to cope with the aligns

Reply:
Ok I understand now. Thanks!